sequencer/batchbuilder.go

```diff
@@ -69,11 +69,29 @@ func (s *Sequencer) tryToProcessTx(ctx context.Context, ticker *time.Ticker) {
 	for _, tx := range s.pendingTxs {
 		log.Infof("processing tx: %s", tx.Hash())
 	}
-	processedTxs, processedTxsHashes, unprocessedTxs, err := s.processTxs(ctx, s.pendingTxs)
+	processedTxs, processedTxsHashes, unprocessedTxs, unprocessedBatch, err := s.processTxs(ctx, s.pendingTxs)
 	if err != nil {
 		log.Errorf("failed to process txs, err: %w", err)
 		return
 	}
+
+	closeBatch := unprocessedBatch
+
+	for unprocessedBatch {
+		// The entire batch hasn't been processed
+		// processedTxs must be reprocessed as the batch was discarded by the executor
+		reprocessTxs := make([]*pool.Transaction, len(processedTxs))
```

Comment:
In this case slice looks like this: `len processedTxs = 2 [*pool.Transaction{}, *pool.Transaction{}, appendedTxs...]`. You should init slice with capacity, not with length.
```suggestion
		reprocessTxs := make([]*pool.Transaction, 0, len(processedTxs))
```

Reply:
Hi!

I did this change but I don't really understand what the problem is not initializaing the capacity in this case, as we are sure of the final length of the slice. In fact I would init it to a len(processTxs) value instead of 0. 

Can you give a more in deep explanation for this?

Thanks!

Reply:
You could see the behavior here - https://go.dev/play/p/wSoQEzIrEhf